### PR TITLE
Fog move to pixel shader

### PR DESCRIPTION
**Pull request by medievil1**

### CODE REVIEW

File: src/core/hle/D3D8/Direct3D9/CxbxPixelShaderTemplate.hlsl [hunk 1]
```diff
@@ -334,6 +340,31 @@ float3 DoBumpEnv(const float4 TexCoord, const float4 BumpEnvMat, const float4 sr
...
+       fogFactor = fogDepth;
+    if(fogTableMode == FOG_TABLE_EXP) 
+       fogFactor = 1 / exp(fogDepth * fogDensity); // 1 / e^(d * density)
+    if(fogTableMode == FOG_TABLE_EXP2) 
+       fogFactor = 1 / exp(pow(fogDepth * fogDensity, 2)); // 1 / e^((d * density)^2)
+    if(fogTableMode == FOG_TABLE_LINEAR) 
+       fogFactor = (fogEnd - fogDepth) / (fogEnd - fogStart);
+	if (FOGENABLE == 0)
+	   fogFactor = 1;
+
```
Comment (jackchentwkh) [inline src/core/hle/D3D8/Direct3D9/CxbxPixelShaderTemplate.hlsl]: I am not sure how flow control in HLSL should be. could this be revised to    
    if (FOGENABLE == 0)
    {
	   fogFactor = 1;
    }
    else
    {
       if(fogTableMode == FOG_TABLE_NONE) 
          fogFactor = fogDepth;
       if(fogTableMode == FOG_TABLE_EXP) 
          fogFactor = 1 / exp(fogDepth * fogDensity); // 1 / e^(d * density)
       if(fogTableMode == FOG_TABLE_EXP2) 
          fogFactor = 1 / exp(pow(fogDepth * fogDensity, 2)); // 1 / e^((d * density)^2)
       if(fogTableMode == FOG_TABLE_LINEAR) 
          fogFactor = (fogEnd - fogDepth) / (fogEnd - fogStart);
    }

File: src/devices/video/EmuNV2A_PGRAPH.cpp [hunk 2]
```diff
@@ -4815,8 +4815,8 @@ int pgraph_handle_method(
                         if (method_count==2){
```
Comment (jackchentwkh) [inline src/devices/video/EmuNV2A_PGRAPH.cpp]: the if (method_count==2){ in line 4815 could be removed as well, remember to remove the other "}" associated with this if statement as well.
there are only two vertex shader modes in PB, program and fixed mode. 
I would suggest to removed all unused code in this section and keep these two lines 
 g_NV2A_VertexShaderMode = VertexShaderMode::ShaderProgram;
g_NV2AVertexShader_dirty = true;

File: src/devices/video/EmuNV2A_PGRAPH.cpp [hunk 3]
```diff
@@ -4804,66 +4804,9 @@ int pgraph_handle_method(
...
-                                //}
-                            }
-						}
-						/*
-						fix function setup finished using 
-						push (NV097_SET_TRANSFORM_EXECUTION_MODE,
-								 NV097_SET_TRANSFORM_EXECUTION_MODE_MODE_FIXED|NV097_SET_TRANSFORM_EXECUTION_MODE_RANGE_MODE_PRIV, method count 1)
-						to start fix function vertex shader program
-						*/
 					}else if (slot== NV097_SET_TRANSFORM_EXECUTION_MODE_MODE_FIXED){//fix function mode
```
Comment (jackchentwkh) [inline src/devices/video/EmuNV2A_PGRAPH.cpp]: please change line 4810 from 
}else if (slot== NV097_SET_TRANSFORM_EXECUTION_MODE_MODE_FIXED){//fix function mode
to 
}else{
  we don't need the test for  if (slot== NV097_SET_TRANSFORM_EXECUTION_MODE_MODE_FIXED) here any more.

File: src/core/hle/D3D8/Direct3D9/CxbxPixelShaderTemplate.hlsl (hunk 1 above)
Comment (jackchentwkh) [reply src/core/hle/D3D8/Direct3D9/CxbxPixelShaderTemplate.hlsl]: @medievil1 I am asking about flow control, not the formatting. did you try the flow control I listed above?

File: src/devices/video/EmuNV2A_PGRAPH.cpp (hunk 3 above)
Comment (medievil1) [reply src/devices/video/EmuNV2A_PGRAPH.cpp]: done....

File: src/core/hle/D3D8/Direct3D9/CxbxPixelShaderTemplate.hlsl (hunk 1 above)
Comment (medievil1) [reply src/core/hle/D3D8/Direct3D9/CxbxPixelShaderTemplate.hlsl]: changed and committed, though did not see any noticeable differences

**Pull request closed without merging**